File: chrome/test/data/webui/bluetooth_internals/bluetooth_internals_browsertest.js
// Copyright 2016 The Chromium Authors
// Use of this source code is governed by a BSD-style license that can be
// found in the LICENSE file.

/**
 * @fileoverview Tests for chrome://bluetooth-internals
 */

GEN('#include "content/public/test/browser_test.h"');

/**
 * Test fixture for BluetoothInternals WebUI testing.
 * @constructor
 * @extends testing.Test
 */
function BluetoothInternalsTest() {}

BluetoothInternalsTest.prototype = {
  __proto__: testing.Test.prototype,

  /** @override */
  browsePreload:
      'chrome://bluetooth-internals/bluetooth_internals.html?module=bluetooth_internals/bluetooth_internals_test.js',

  /** @override */
  isAsync: true,

  /** @override */
  extraLibraries: [
    '//third_party/mocha/mocha.js',
    '//chrome/test/data/webui/mocha_adapter.js',
  ],
};

TEST_F('BluetoothInternalsTest', 'Startup_BluetoothInternals', function() {
  // Run all registered tests.
  mocha.run();
});

File: chrome/test/data/webui/bluetooth_internals/bluetooth_internals_browsertest.cc
// Copyright 2023 The Chromium Authors
// Use of this source code is governed by a BSD-style license that can be
// found in the LICENSE file.

#include "base/command_line.h"
#include "chrome/common/webui_url_constants.h"
#include "chrome/test/base/web_ui_mocha_browser_test.h"
#include "content/public/common/content_switches.h"
#include "content/public/test/browser_test.h"

class BluetoothInternalsTest : public WebUIMochaBrowserTest {
 protected:
  BluetoothInternalsTest() {
    set_test_loader_host(std::string(chrome::kChromeUIBluetoothInternalsHost) +
                         "?isTest");
  }

  void SetUpCommandLine(base::CommandLine* command_line) override {
    WebUIMochaBrowserTest::SetUpCommandLine(command_line);
    // Enables the MojoJSTest bindings which are used in
    // bluetooth_internals_test.js.
    base::CommandLine::ForCurrentProcess()->AppendSwitchASCII(
        switches::kEnableBlinkFeatures, "MojoJSTest");
  }
};

IN_PROC_BROWSER_TEST_F(BluetoothInternalsTest, Startup_BluetoothInternals) {
  RunTestWithoutTestLoader("bluetooth_internals/bluetooth_internals_test.js",
                           "mocha.run()");
}
